src/typescript-service.ts

```diff
@@ -1590,33 +1604,41 @@ export class TypeScriptService {
 			.map(config => {
 				const symbols: SymbolInformation[] = [];
 				config.ensureAllFiles();
-				if (query) {
-					const items = config.getService().getNavigateToItems(query, undefined, undefined, false);
-					for (const item of items) {
-						const si = this._transformNavItem(this.root, config.getProgram(), item);
-						if (!util.isLocalUri(si.location.uri)) {
-							continue;
-						}
-						symbols.push(si);
-					}
-				} else if (symQuery) {
-					// TODO(beyang): after workspace/symbol extension is accepted into LSP, push this logic upstream to getNavigateToItems
-					const items = config.getService().getNavigateToItems(symQuery.name || '', undefined, undefined, false);
-					const packageName = config.getPackageName();
-					const pd = packageName ? { name: packageName } : undefined;
-					for (const item of items) {
-						const sd = SymbolDescriptor.create(item.kind, item.name, item.containerKind, item.containerName, pd);
-						if (!util.symbolDescriptorMatch(symQuery, sd)) {
-							continue;
+				const program = config.getProgram();
+				if (!program) {
+					return symbols;
+				}
+				try {
+					if (query) {
+						const items = config.getService().getNavigateToItems(query, undefined, undefined, false);
+						for (const item of items) {
+							const si = this._transformNavItem(this.root, program, item);
+							if (!util.isLocalUri(si.location.uri)) {
+								continue;
+							}
+							symbols.push(si);
 						}
-						const si = this._transformNavItem(this.root, config.getProgram(), item);
-						if (!util.isLocalUri(si.location.uri)) {
-							continue;
+					} else if (symQuery) {
+						// TODO(beyang): after workspace/symbol extension is accepted into LSP, push this logic upstream to getNavigateToItems
+						const items = config.getService().getNavigateToItems(symQuery.name || '', undefined, undefined, false);
+						const packageName = config.getPackageName();
+						const pd = packageName ? { name: packageName } : undefined;
+						for (const item of items) {
+							const sd = SymbolDescriptor.create(item.kind, item.name, item.containerKind, item.containerName, pd);
+							if (!util.symbolDescriptorMatch(symQuery, sd)) {
+								continue;
+							}
+							const si = this._transformNavItem(this.root, program, item);
+							if (!util.isLocalUri(si.location.uri)) {
```

Comment:
The behavior of this function is not correct, it checks the scheme not the host. What is this check for?
(I realised this is just reindented, but noticed anyway)